flexmeasures/data/schemas/generic_assets.py

```diff
@@ -151,16 +152,24 @@ class Meta:
 class GenericAssetIdField(MarshmallowClickMixin, fields.Int):
     """Field that deserializes to a GenericAsset and serializes back to an integer."""
 
-    @with_appcontext_if_needed()
-    def _deserialize(self, value, attr, obj, **kwargs) -> GenericAsset:
+    def __init__(self, status_if_not_found: HTTPStatus | None = None, *args, **kwargs):
+        self.status_if_not_found = status_if_not_found
+        super().__init__(*args, **kwargs)
+
+    def _deserialize(self, value: int, attr, obj, **kwargs) -> GenericAsset:
```

Comment:
`value` arg doesn't have the right type considering that you are using `int(value)` to convert the object to `int`, is it `str`?